ska_helpers/utils.py

```diff
@@ -371,3 +372,91 @@ def convert_to_int_float_str(val: str) -> int | float | str:
                 out = val
 
     return out
+
+
+class TypedDescriptor:
+    """Class to create a descriptor for an attribute that is cast to a type.
+
+    This is a base class for creating a descriptor that can be used to define an
+    attribute on a class that is cast to a specific type.  The type is specified by
+    setting the ``cls`` class attribute on the descriptor class.
+
+    Most commonly ``cls`` is a class like ``CxoTime`` or ``Quat``, but it could also
+    be a function like ``int`` or ``float``.
```

Comment:
> function like ``int`` or ``float``

do you mean _a type like ``int`` or ``float``_?

Reply:
After I thought about it, I understood what you meant. You meant a _factory function_ to construct the instances. I would still rephrase it.

Reply:
Done